components/director/pkg/operation/update_operation_handler.go

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2020 The Compass Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package operation
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/persistence"
+
+	"github.com/go-ozzo/ozzo-validation/is"
+
+	validation "github.com/go-ozzo/ozzo-validation"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/resource"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+)
+
+// OperationRequest is the expected request body when updating certain operation status
+type OperationRequest struct {
+	OperationType graphql.OperationType `json:"operation_type,omitempty"`
+	ResourceType  resource.Type         `json:"resource_type"`
+	ResourceID    string                `json:"resource_id"`
+	Error         string                `json:"error"`
+}
+
+// ResourceUpdaterFunc defines a function which updates a particular resource ready and error status
+type ResourceUpdaterFunc func(ctx context.Context, id string, ready bool, errorMsg *string) error
+
+// ResourceFetcherFunc defines a function which deletes a particular resource by ID
+type ResourceDeleterFunc func(ctx context.Context, id string) error
+
+type updateOperationHandler struct {
+	transact             persistence.Transactioner
+	resourceUpdaterFuncs map[resource.Type]ResourceUpdaterFunc
+	resourceDeleterFuncs map[resource.Type]ResourceDeleterFunc
+}
+
+// NewUpdateOperationHandler creates a new handler struct to update resource by operation
+func NewUpdateOperationHandler(transact persistence.Transactioner, resourceUpdaterFuncs map[resource.Type]ResourceUpdaterFunc, resourceDeleterFuncs map[resource.Type]ResourceDeleterFunc) *updateOperationHandler {
+	return &updateOperationHandler{
+		transact:             transact,
+		resourceUpdaterFuncs: resourceUpdaterFuncs,
+		resourceDeleterFuncs: resourceDeleterFuncs,
+	}
+}
+
+// ServeHTTP handles the Operations API requests
+func (h *updateOperationHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
```

Comment:
Am I missing something or does the writer never get used to set and return appropriate status code (+ response body? empty). Probably 200 OK is the default but maybe it would be good to be explicit and set it manually

Reply:
you are right, it is missing